src/components/dialogs/filter/filter-properties.tsx

```diff
@@ -0,0 +1,213 @@
+                            propertyValuesTest(values, context, false)
+                    ),
+                [PROPERTY_VALUES_1]: yup
+                    .array()
+                    .of(yup.string())
+                    .test(
+                        'can not be empty if line',
+                        'YupRequired',
+                        (values, context) =>
+                            propertyValuesTest(values, context, true)
+                    ),
+                [PROPERTY_VALUES_2]: yup
+                    .array()
+                    .of(yup.string())
+                    .test(
+                        'can not be empty if line',
+                        'YupRequired',
+                        (values, context) =>
+                            propertyValuesTest(values, context, true)
+                    ),
+            })
+        )
+        .test(
+            'distinct names',
+            'filterPropertiesNameUniquenessError',
+            (properties, context) => {
+                // with context.from[length - 1], we can access to the root fields of the form
+                const rootLevelForm = context!.from![context!.from!.length - 1];
+                const filterType = rootLevelForm.value[FILTER_TYPE];
+                if (filterType !== FilterType.CRITERIA_BASED.id) {
+                    // we don't test if we are not in a criteria based form
+                    return true;
+                }
+                const names = properties! // never null / undefined
+                    .filter((prop) => !!prop[PROPERTY_NAME])
+                    .map((prop) => prop[PROPERTY_NAME]);
+                return areArrayElementsUnique(names);
+            }
+        ),
+};
+
+function fetchPredefinedProperties() {
+    return fetchAppsAndUrls().then((res) => {
+        const studyMetadata = res.find(
+            (metadata: any) => metadata.name === 'Study'
+        );
+        if (!studyMetadata) {
+            return Promise.reject(
+                'Study entry could not be found in metadatas'
+            );
+        }
+
+        return studyMetadata?.predefinedEquipmentProperties?.substation;
+    });
+}
+
+function FilterProperties() {
+    const { snackError } = useSnackMessage();
+
+    const watchEquipmentType = useWatch({
+        name: EQUIPMENT_TYPE,
+    });
+    const [fieldProps, setFieldProps] = useState({});
+
+    const fieldName = `${CRITERIA_BASED}.${FILTER_PROPERTIES}`;
+
+    const {
+        fields: filterProperties, // don't use it to access form data ! check doc,
+        append,
+        remove,
+    } = useFieldArray({
+        name: fieldName,
+    });
+
+    const isForLineOrHvdcLine =
+        watchEquipmentType === Line.type || watchEquipmentType === Hvdc.type;
+
+    const isForSubstation = watchEquipmentType === Substation.type;
+
+    function addNewProp() {
+        if (isForLineOrHvdcLine) {
+            append({
+                [PROPERTY_NAME]: '',
```

Comment:
```suggestion
                [PROPERTY_NAME]: null,
```
=> it's a select input, default empty value should be null to prevent "delete button" from appearing

Reply:
I think I just kept the previous default value, you're right I will change it